### PR TITLE
fix paste(collapse='') lints inside pipes

### LINKED ISSUE #2571: `paste_linter()` doesn't work as expected with pipes
``` r
library(lintr)

lint(text = 'paste0(raw, collapse = "")', linters = paste_linter())
#> <text>:1:1: warning: [paste_linter] Use paste(), not paste0(), to collapse a character vector when sep= is not used.
#> paste0(raw, collapse = "")
#> ^~~~~~~~~~~~~~~~~~~~~~~~~~
lint(text = 'raw %>% paste0(collapse = "")', linters = paste_linter())
```

<sup>Created on 2024-05-05 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #2571

### CODE REVIEW

File: tests/testthat/test-paste_linter.R [hunk 1]
```diff
@@ -249,6 +249,16 @@ test_that("paste0(collapse=...) is caught", {
   expect_lint("paste0(foo(x, y, z), collapse = '')", lint_msg, linter)
 })
 
+test_that("paste0(collapse=...) is caught in pipes", {
+  linter <- paste_linter()
+  lint_msg <- rex::rex("Use paste(), not paste0(), to collapse a character vector when sep= is not used.")
+
+  expect_lint('x %>% paste0(y, collapse = "")', NULL, linter)
```
Comment (IndrajeetPatil) [inline tests/testthat/test-paste_linter.R:256]: There is a discrepancy here between pipes for code that produces identical results:

``` r
library(magrittr)
library(lintr)

x <- letters[1:2]
x <- x %>% paste0(x, collapse = "")
lint(text = 'x <- x %>% paste0(x, collapse = "")', linters = paste_linter())


x <- letters[1:2]
x %<>% paste0(x, collapse = "")
lint(text = 'x %<>% paste0(collapse = "")', linters = paste_linter())
#> <text>:1:8: warning: [paste_linter] Use paste(), not paste0(), to collapse a character vector when sep= is not used.
#> x %<>% paste0(collapse = "")
#>        ^~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2024-05-06 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Comment (MichaelChirico) [reply tests/testthat/test-paste_linter.R:256]: Maybe just a typo? The code in the latter example doesn't match the linted code.

Comment (IndrajeetPatil) [reply tests/testthat/test-paste_linter.R:256]: Ah, so dumb 🤦 

Anyway, should we include test for other pipes? I don't think it's needed, but just wanted to check.

``` r
library(lintr)
lint(text = 'x %<>% paste0(collapse = "")', linters = paste_linter())
#> <text>:1:8: warning: [paste_linter] Use paste(), not paste0(), to collapse a character vector when sep= is not used.
#> x %<>% paste0(collapse = "")
#>        ^~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2024-05-06 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Comment (MichaelChirico) [reply tests/testthat/test-paste_linter.R]: OK, SG; done.

**Review by IndrajeetPatil — APPROVED** [review]

**Pull request merged**